### PR TITLE
ci: enable windows 'sanitycheck --build-only'

### PR DESCRIPTION
**Pull request by thst-nordic**
- Small tweaks to fix paths/memory allocation.
- Use make from chocolatey

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -1404,7 +1398,7 @@ class MakeGenerator:
                         goal.fail("unknown_error")
 
                 else:
-                    goal = self.goals[name]
+                    goal = self.goals[name.strip()]
```
Comment (thst-nordic) [inline scripts/sanitycheck]: extra windows line ending needs to be removed: '\r'

File: scripts/sanitycheck [hunk 2]
```diff
@@ -1800,6 +1793,15 @@ class TestCase:
                 matches = [ match.decode().replace("test_", "") for match in _matches ]
                 return matches, warnings
 
+            if os.name == 'nt':
+                with contextlib.closing(mmap.mmap(inf.fileno(), 0, access=mmap.ACCESS_READ)) as main_c:
+                    return mmap_worker(main_c)
```
Comment (thst-nordic) [inline scripts/sanitycheck]: [mmap on windows doesn't support MAP_PRIVATE. ](https://docs.python.org/3.0/library/mmap.html)

File: scripts/sanitycheck [hunk 3]
```diff
@@ -3208,6 +3210,16 @@ def main():
     global run_individual_tests
     options = parse_arguments()
 
+    if os.name == 'nt':
+        if (not options.build_only) or (not options.disable_size_report):
+            print("Testing with sanitycheck on Windows is not fully supported yet.")
+            print("Only build without size report works for now:")
+            print("\t- '--build-only --disable-size-report'")
+            print("\t- Requires make: 'choco install make'")
+            print("\t- May require long-path support in Windows OS.")
```
Comment (thst-nordic) [inline scripts/sanitycheck]: sample paths are under 256, but the test paths are mostly >256

File: scripts/sanitycheck [hunk 4]
```diff
@@ -1771,11 +1765,10 @@ class TestCase:
         achtung_regex = re.compile(
             br"(#ifdef|#endif)",
             re.MULTILINE)
-        warnings = None
 
         with open(inf_name) as inf:
-            with contextlib.closing(mmap.mmap(inf.fileno(), 0, mmap.MAP_PRIVATE,
```
Comment (carlescufi) [inline scripts/sanitycheck]: Wouldn't it be simpler to create a variable mmap_flags that you set conditionally for UNIX or Windows and then keep the code as-is without having to move it to a nested function and having to have 2 calls to mmap()?

File: scripts/sanitycheck [hunk 5]
```diff
@@ -3208,6 +3210,16 @@ def main():
     global run_individual_tests
     options = parse_arguments()
 
+    if os.name == 'nt':
+        if (not options.build_only) or (not options.disable_size_report):
```
Comment (carlescufi) [inline scripts/sanitycheck]: nitpicking, but why not have a single `if X and Y` instead of 2 `if` statements?

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: scripts/sanitycheck (hunk 4 above)
Comment (thst-nordic) [reply scripts/sanitycheck]: yes, I'll update.

File: scripts/sanitycheck (hunk 5 above)
Comment (thst-nordic) [reply scripts/sanitycheck]: I can update it.  

print("Testing with sanitycheck on Windows is not fully supported yet.")
 ^^ was above the 2nd if before I moved it upstream.

**Review by SebastianBoe — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@nashif can you please take a look?

**Review by nashif — APPROVED** [review]

File: scripts/sanitycheck [hunk 6]
```diff
@@ -3317,6 +3322,16 @@ def main():
         error("west-flash requires device-testing to be enabled")
         sys.exit(1)
 
+    if os.name == 'nt' and \
+       (not options.build_only or not options.disable_size_report):
+        print("Testing with sanitycheck on Windows is not fully supported yet.")
+        print("Only build without size report works for now:")
+        print("\t- '--build-only --disable-size-report'")
```
Comment (marc-hb) [inline scripts/sanitycheck]: I don't think burying the start of Windows documentation in just one catch-all Windows error message is a good idea. I think it's better and more "scalable" to keep Windows-specific tweaks separate from each other and generally closer to the corresponding code. This also allows "smarter" behaviour, here for instance.

```python
if os.name == 'nt' and not options.disable_size_report:
     warning ("Windows doesn't support size report yet, forcing --disable-size-report")
     options.disable_size_report = true
```
Just a suggestion/example of something not really possible with a grab bag error message.

File: scripts/sanitycheck [hunk 7]
```diff
@@ -3317,6 +3322,16 @@ def main():
...
+        print("\t- Requires make: 'choco install make'")
```
Comment (marc-hb) [inline scripts/sanitycheck]: I don't see why an installation tip for make is buried in an error message about unrelated options.

You already updated the documentation which is great. Now what is the error message for people who already installed their environment earlier and miss make? If it's something like "make not found" then there's nothing to do, you can just remove that line. If the error is more cryptic for some unknown reason then implement a better error message just before the first make invocation:

```python
if make_not_found:
    error("make not found, check the getting started guide")
```

File: scripts/sanitycheck [hunk 8]
```diff
@@ -3317,6 +3322,16 @@ def main():
...
+        print("\t- May require long-path support in Windows OS.")
```
Comment (marc-hb) [inline scripts/sanitycheck]: Similar comment: I don't see why this  information is tied to --build-only and --disable-size-report (and missing from issue #2664?)

**Review by marc-hb — CHANGES_REQUESTED** [review]
Please reference issue #2664 in the commit message to enable more github magic and help people find more information.

**Comment by thst-nordic** [comment]
closing due to refactor.

**Comment by marc-hb** [comment]
> closing due to refactor.

For the record: PR #18603

**Pull request closed without merging**